packages/inspector/src/components/EntityInspector/GltfNodeModifiersInspector/GltfNodeModifiersInspector.tsx

```diff
@@ -0,0 +1,230 @@
+import { PbrMaterial } from '../MaterialInspector/PbrMaterial';
+import { Texture } from '../MaterialInspector/Texture/types';
+import type { Props } from './types';
+import { ensureTextureDefaults, fromComponent, isValidInput, toComponent } from './utils';
+
+import './GltfNodeModifiersInspector.css';
+
+export default withSdk<Props>(({ sdk, entity, initialOpen = true }) => {
+  const files = useAppSelector(selectAssetCatalog);
+  const { GltfNodeModifiers } = sdk.components;
+
+  const hasComponent = useHasComponent(entity, GltfNodeModifiers);
+  const { getInputProps } = useComponentInput(
+    entity,
+    GltfNodeModifiers,
+    fromComponent(files?.basePath ?? ''),
+    toComponent(files?.basePath ?? ''),
+    isValidInput,
+  );
+
+  const [componentValue] = useComponentValue<PBGltfNodeModifiers>(entity, GltfNodeModifiers);
+
+  const handleRemove = useCallback(async () => {
+    sdk.operations.removeComponent(entity, GltfNodeModifiers);
+    await sdk.operations.dispatch();
+  }, [entity, GltfNodeModifiers]);
+
+  const addSwap = () => {
+    const current = componentValue ?? ({ modifiers: [] } as PBGltfNodeModifiers);
+    const existing = fromComponent(files?.basePath ?? '')(current).swaps;
+    const newSwaps = [
+      ...existing,
+      {
+        path: '',
+        castShadows: true,
+        material: ensureTextureDefaults({
+          type: MaterialType.MT_PBR,
+          texture: {
+            type: Texture.TT_TEXTURE,
+            src: '',
+            wrapMode: '0',
+            filterMode: '0',
+            offset: { x: '0', y: '0' },
+            tiling: { x: '1', y: '1' },
+          },
+          bumpTexture: {
+            type: Texture.TT_TEXTURE,
+            src: '',
+            wrapMode: '0',
+            filterMode: '0',
+            offset: { x: '0', y: '0' },
+            tiling: { x: '1', y: '1' },
+          },
+          emissiveTexture: {
+            type: Texture.TT_TEXTURE,
+            src: '',
+            wrapMode: '0',
+            filterMode: '0',
+            offset: { x: '0', y: '0' },
+            tiling: { x: '1', y: '1' },
+          },
+        } as MaterialInput),
+      },
+    ];
+    getInputProps('swaps').onChange?.({ target: { value: newSwaps } } as any);
+  };
+
+  const swapsValue = useMemo(() => {
+    const raw = (getInputProps('swaps').value as any) ?? [];
+    const coerce = (val: any) => {
+      if (Array.isArray(val)) return val;
+      if (val && typeof val === 'object') {
+        const keys = Object.keys(val)
+          .filter(k => /^\d+$/.test(k))
+          .sort((a, b) => Number(a) - Number(b));
+        return keys.map(k => val[k]);
+      }
+      return [] as any[];
+    };
+    const fromInput = coerce(raw);
+    if (fromInput.length > 0) return fromInput;
+    const current = componentValue ?? ({ modifiers: [] } as PBGltfNodeModifiers);
+    return fromComponent(files?.basePath ?? '')(current).swaps;
+  }, [componentValue, files, getInputProps]);
```

Comment:
I think the `coerce` processing is unnecessary since the `GltfNodeModifier` returns an array.

```suggestion
const swapsValue = useMemo(() => {
  const swaps = (getInputProps('swaps').value as unknown[] | undefined) ?? [];
  if (swaps.length > 0) return swaps;

  const current: PBGltfNodeModifiers = componentValue ?? { modifiers: [] };
  return fromComponent(files?.basePath ?? '')(current).swaps;
}, [componentValue, files?.basePath, getInputProps]);
```